options.go

```diff
@@ -89,10 +97,16 @@ func DeviceIDRegexMustCompile() Option {
 
 type deviceIDRegexMustCompileOption struct{}
 
-func (deviceIDRegexMustCompileOption) Validate(val Validator) error {
-	if err := val.ValidateDeviceId(); err != nil {
-		return err
+func (deviceIDRegexMustCompileOption) Validate(i any) error {
+	switch r := i.(type) {
+	case *RegistrationV1:
+		return r.ValidateDeviceId()
+	case *RegistrationV2:
+		//Matcher description is for Events. Are we not matching for DeviceId in Reg2?
```

Comment:
We are but we're exposing the regex to the entire wrp message, so the user won't be limited to only `DeviceId`.

let's just throw in
```suggestion
		return fmt.Errorf("%w: RegistrationV2 does not use DeviceID directly, use `FieldRegex` instead", ErrInvalidType)
```

Reply:
if it's using `FieldRegex`, instead of throwing an error should we use the `ValidateEventRegex` function? and rename that function for V2 something like `ValidateFieldRegex`?

Reply:
or should we create a new `FieldRegexOption` to be used for V2 events and device ID ?

Reply:
well validation has to occur on both fields `Field` and `Regex`.

```go
// FieldRegex is a substructure with data related to regular expressions.
type FieldRegex struct {
	// Field is the wrp field to be used for regex.
	// All wrp field can be used, refer to the schema for examples.
	Field string `json:"field"`

	// FieldRegex is the regular expression to match `Field` against to.
	Regex string `json:"regex"`
}
```

I recommend leaving the error and updating the error message and create a ticket outlining the validation we need for `FieldRegex`. Again `FieldRegex` opens up the entire wrp message (not just device id).

Reply:
let's implement it in a later pr